ui/app/AppLayouts/Chat/views/ChatMessagesView.qml

```diff
@@ -62,7 +62,7 @@ Item {
                 return
             }
 
-            if (chatDetails && chatDetails.active && chatDetails.hasUnreadMessages && !messageStore.loading) {
+            if (chatDetails && chatDetails.active && (chatDetails.hasUnreadMessages || chatDetails.highlight) && !messageStore.loading) {
```

Comment:
Wouldn't it be enough to just use `chatDetails.highlight`?

Reply:
`highlight` is only/mostly for when the chat is newly created. It doesn't get set to `true` when there are new messages.

In our UI, `highlight: true` and `hasUnreadMessages: true` cause the same visual effect of putting the name of the chat bold, but that's just our design. I think mobile is more elegant and has a different effect for that. This is why the bug appeared. It looks like we have unread messages but we don't.

Reply:
> highlight is only/mostly for when the chat is newly created.

Ah, okay, that is very strange logic then. IMHO, we should only rely on the highlight property and hide all the logic behind it, whether it's a new chat or if it has unread messages, etc. That would require status-go to be fixed, but I believe we will benefit from it by having all platforms' chats highlighted under the same conditions.

Reply:
> Ah, okay, that is very strange logic then. IMHO, we should only rely on the highlight property and hide all the logic behind it, whether it's a new chat or if it has unread messages, etc.

I would agree with that too, but that's not how `highlight` was imagined it seems.

IMO, it's not that useful or it would be if it was like you said, but it's used in very specific cases like when the chat is created or even when we clear the history:
```go
func (db sqlitePersistence) clearHistory(chat *Chat, currentClockValue uint64, tx *sql.Tx, deactivate bool) error {
        ...
	chat.UnviewedMessagesCount = 0
	chat.UnviewedMentionsCount = 0
	chat.Highlight = true
```
In this case above, your idea doesn't work sadly.

Also, I think mobile has a different visual effect for `unread` and `highlight`.

The only way to clear `highlight` currently is calling mark as read.